### DISCUSSION TITLE
install-windows-zip

### DISCUSSION THREAD

**Discussion by giscus** [brainboxdotcc/dpp-comments#13] [General]
# install-windows-zip

A lightweight C++ Discord API library supporting the entire Discord API, including Slash Commands, Voice/Audio, Sharding, Clustering and more!

https://dpp.dev/install-windows-zip.html

**Comment by AtackYT** [comment]
Any other ways to get the dependencies without using visual studio such as using g++/gcc?

**Reply by braindigitalis** [reply]
you could build your own copy from source. There are so many varieties of compiler, operating system, toolchain etc that we can't possibly hope to directly support them all with documentation and guidance on Discord. For most people "this is the way", and if you want to do it differently you are expected to know what you are doing. Compilation on windows via g++ is certainly not impossible, but not for the newcomer either. All precompiled dependencies are taken from vcpkg. Good luck!
